src/parachain/issue.ts

```diff
@@ -121,33 +159,73 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI
         this.feeAPI = new DefaultFeeAPI(api);
     }
 
+    async getRequestLimits(vaults?: Map<AccountId, Big>): Promise<IssueLimits> {
+        if (!vaults) vaults = await this.vaultsAPI.getVaultsWithIssuableTokens();
+        const [singleVaultMaxIssuable, totalMaxIssuable] = [...vaults.entries()].reduce(
+            ([singleVault, maxTotal], [_, vaultAvailable]) => {
+                maxTotal = maxTotal.plus(vaultAvailable);
+                singleVault = singleVault.gt(vaultAvailable) ? singleVault : vaultAvailable;
+                return [singleVault, maxTotal];
+            },
+            [new Big(0), new Big(0)]
+        );
+        return { singleVaultMaxIssuable, totalMaxIssuable };
+    }
+
     /**
      * @param events The EventRecord array returned after sending an issue request transaction
      * @returns The issueId associated with the request. If the EventRecord array does not
      * contain issue request events, the function throws an error.
      */
-    private getRequestIdFromEvents(events: EventRecord[]): Hash {
-        for (const { event } of events) {
-            if (this.api.events.issue.RequestIssue.is(event)) {
-                const hash = this.api.createType("Hash", event.data[0]);
-                return hash;
+    private getIssueIdsFromEvents(events: EventRecord[]): Hash[] {
+        return getRequestIdsFromEvents(events, this.api.events.issue.RequestIssue, this.api);
+    }
+
+    async request(
+        amount: Big,
+        options?: {
```

Comment:
I think in most places we flattened the options. Any particular reason you want to submit an options object here?

Reply:
Mainly because there's no definite order for the optional arguments. But I think they can actually be flattened by putting the `availableVaults` argument last, that should work for most cases.